File: 02-eda-1/scatter.py
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_csv("movies.csv")
df = df.dropna(subset=["VOTES", "Gross"])
# Convert Votes column from string to int
df["VOTES"] = df["VOTES"].str.replace(",", "").astype(int)
# Convert Gross column from string to float
df["Gross"] = (df["Gross"].str.replace("$", "")).str.replace("M", "").astype(float)

plt.scatter(df["VOTES"], df["Gross"])
plt.xlabel('Votes')
plt.ylabel('Gross (Millions)')
plt.title('Scatter Plot of Votes vs Gross')
plt.grid(True)
plt.show()

File: 02-eda-1/scatter.r
df <- read.csv("movies.csv")
df <- na.omit(df)
# Convert VOTES column from string to int
df$VOTES <- as.integer(gsub(",", "", df$VOTES))
# Convert Gross column from string to float
df$Gross <- as.numeric(gsub("[$M]", "", df$Gross))

# Plot scatter plot
plot(df$VOTES, df$Gross,
     xlab = "Votes",
     ylab = "Gross (Millions)",
     main = "Scatter Plot of Votes vs Gross")

abline(lm(df$VOTES ~ df$Gross), data = df)
